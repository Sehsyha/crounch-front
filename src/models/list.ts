import { Product } from './product';
import { CategoryInGetListResponse } from './category';

interface List {
  id: string
  name: string
  products: Product[]
}

interface GetListResponse {
  id: string
  name: string
  creationDate: string
  categories: CategoryInGetListResponse[]
}

export { List, GetListResponse }
