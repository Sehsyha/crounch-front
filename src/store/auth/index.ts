import { Module } from 'vuex';

import { TOKEN_STORAGE_KEY } from '@/utils/constants';
import { getters } from './getters';
import { actions } from './actions';
import { mutations } from './mutations';
import { RootState } from '..';

export interface AuthState {
  token: string;
  status: string;
}

export const state: AuthState = {
  token: localStorage.getItem(TOKEN_STORAGE_KEY) || '',
  status: '',
};

const namespaced: boolean = true;
export const authNamespace = { namespace: 'auth' };

export const auth: Module<AuthState, RootState> = {
  namespaced,
  state,
  getters,
  actions,
  mutations,
};
