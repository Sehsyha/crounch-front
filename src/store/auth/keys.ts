export enum AuthActions {
  LOGIN = 'LOGIN',
  SIGNUP = 'SIGNUP',
  LOGOUT = 'LOGOUT',
}

export enum AuthGetters {
  IS_AUTHENTICATED = 'IS_AUTHENTICATED',

}

export enum AuthMutations {
  LOGIN = 'LOGIN',
  LOGOUT = 'LOGOUT',
}
