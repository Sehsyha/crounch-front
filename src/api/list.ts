import { doFetch } from './api';
import { List } from '@/models/list';

export function createList(name: string): Promise<List> {
  return doFetch({
    url: 'lists',
    method: 'POST',
    data: { name }
  });
}

export function getOwnerLists(): Promise<Array<List>> {
  return doFetch({
    url: 'lists',
    method: 'GET',
  });
}

export function deleteList(id: string): Promise<void> {
  return doFetch({
    url: `lists/${id}`,
    method: 'DELETE',
  });
}

export function addProductToList(productID: string, listID: string): Promise<void> {
  return doFetch({
    url: `lists/${listID}/products/${productID}`,
    method: 'POST'
  })
}

export function readList(listID: string): Promise<List> {
  return doFetch({
    url: `lists/${listID}`,
    method: 'GET'
  })
}
