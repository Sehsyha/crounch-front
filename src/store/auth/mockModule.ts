import { AuthKeys } from './keys';
import { Module } from 'vuex';
import { AuthState } from '.';
import { RootState } from '..';

const actions = {
  [AuthKeys.SIGNUP]: jest.fn(),
  [AuthKeys.LOGIN]: jest.fn(),
  [AuthKeys.LOGOUT]: jest.fn(),
};

const getters = {
  isAuthenticated: jest.fn(),
};

export function createAuthModuleMock(): Module<AuthState, RootState> {
  return {
    namespaced: true,
    actions,
    getters,
    state: {
      token: '',
      status: '',
    },
  };
}
