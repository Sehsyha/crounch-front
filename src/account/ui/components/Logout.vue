<template>
  <button id="logout" @click.prevent="logout">Se dÃ©connecter</button>
</template>

<script lang="ts">
import Vue from 'vue';
import { Component } from 'vue-property-decorator';
import { getModule } from 'vuex-module-decorators';

import { ListModule } from '@/store/list/ListModule';
import { AccountModule } from '@/account/store/AccountModule';

@Component
export default class Logout extends Vue {
  public accountModule: AccountModule = getModule(AccountModule)
  public listModule: ListModule = getModule(ListModule)

  logout() {
    this.listModule.reset()
    this.accountModule.logoutAction().then(() => {
      this.$router.push('/');
    })
  }
}
</script>
