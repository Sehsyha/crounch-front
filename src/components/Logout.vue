<template>
  <button id="logout" @click.prevent="logout">Se dÃ©connecter</button>
</template>

<script lang="ts">
import Vue from 'vue';
import { Component } from 'vue-property-decorator';
import { getModule } from 'vuex-module-decorators';

import { ListModule } from '@/store/list/ListModule';
import { AuthModule } from '@/store/AuthModule';

@Component
export default class Logout extends Vue {
  public authModule: AuthModule = getModule(AuthModule)
  public listModule: ListModule = getModule(ListModule)

  logout() {
    this.listModule.reset()
    this.authModule.logoutAction().then(() => {
      this.$router.push('/');
    })
  }
}
</script>
