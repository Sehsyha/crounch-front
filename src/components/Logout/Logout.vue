<template>
  <button id="logout" @click.prevent="logout">Se d√©connecter</button>
</template>

<script lang="ts">
import Vue from 'vue';
import { Component } from 'vue-property-decorator';
import { Action, Mutation } from 'vuex-class';

import { AuthKeys } from '@/store/auth/keys';
import { authNamespace } from '@/store/auth';
import { ListKeys } from '../../store/list/keys';
import { listNamespace } from '../../store/list';

@Component
export default class Logout extends Vue {
  @Action(AuthKeys.LOGOUT, authNamespace) doLogout: any;
  @Mutation(ListKeys.RESET, listNamespace) doReset: any;

  logout() {
    this.doReset();
    this.doLogout();
    this.$router.push('/');
  }
}
</script>
