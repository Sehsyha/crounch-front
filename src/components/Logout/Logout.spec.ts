import { Wrapper } from '@vue/test-utils';
import { Module } from 'vuex';

import { AuthState } from '@/store/auth';
import { RootState } from '@/store';
import { createAuthModuleMock } from '@/store/auth/mockModule';
import { shallowComponent } from '@/utils/test';
import Logout from './Logout.vue';
import { AuthActions } from '@/store/auth/keys';
import { ListMutations } from '@/store/list/keys';
import { ListState } from '@/store/list';
import { createListModuleMock } from '@/store/list/mockModule';

describe('Logout', () => {
  let wrapper: Wrapper<Logout>;
  let auth: Module<AuthState, RootState>;
  let list: Module<ListState, RootState>;

  beforeEach(() => {
    auth = createAuthModuleMock();
    list = createListModuleMock();

    const modules = {
      auth,
      list,
    };

    wrapper = shallowComponent(Logout, { modules });
  });

  it('Should render.', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should call logout when logout button is clicked.', () => {
    wrapper.find('button').trigger('click');

    expect(auth.actions![AuthActions.LOGOUT]).toHaveBeenCalled();
  });

  it('Should call reset mutation when logout button is clicked.', () => {
    wrapper.find('button').trigger('click');

    expect(list.mutations![ListMutations.RESET]).toHaveBeenCalled();
  });

  it('Should redirect to home page when button is clicked.', () => {
    wrapper.find('button').trigger('click');

    expect(wrapper.vm.$router.push).toHaveBeenCalledWith('/');
  });
});
