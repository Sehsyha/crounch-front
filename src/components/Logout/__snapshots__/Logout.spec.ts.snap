// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout Should render. 1`] = `<button id="logout">Se d√©connecter</button>`;
